### PR TITLE
Add `AI Text` block

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Introduces a new block which uses the Service Module to request AI text completion from OpenAI. 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203358502199087/1203553091426343/f) _(internal)_

## 🔍 What does this change?

- Adds the block

## 🐾 Next steps

- Once approved, recreate the type in production and update this PR
- Test the block again in WordPress
- Merge
- Publish

**Follow up**
- Allow users to select the model (internal [task](https://app.asana.com/0/1203358502199087/1203701786066059/f))

## ❓ How to test this?

1.  `yarn && cd blocks/ai-text && yarn dev`
2. add a valid Block Protocol API key to `src/dev.tsx`
3. Try generating some text

### CODE REVIEW

**Review by nonparibus — DISMISSED** [review]
Legal ✅

File: blocks/ai-text/src/app/generate-text.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      data: {
+        entityId,
+        entityTypeId,
+        properties: {
+          [contentKey]: generatedText,
+          [modelKey]: model,
+          [promptKey]: promptText,
+        },
+      },
+    });
```
Comment (benwerner01) [inline blocks/ai-text/src/app/generate-text.tsx:88]: Nit: you could use `useCallback` here as well
```suggestion
  const handleConfirm = useCallback(
    () =>
      graphModule.updateEntity({
        data: {
          entityId,
          entityTypeId,
          properties: {
            [contentKey]: generatedText,
            [modelKey]: model,
            [promptKey]: promptText,
          },
        },
      }),
    [graphModule, entityId, entityTypeId, generatedText, model, promptText],
  );
```

**Review by benwerner01 — DISMISSED** [review]
Thanks Ciaran, this is working for me locally and looks good to me. Left 3 minor comments, none of which should block merging.

File: blocks/ai-text/src/dev.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * 3. It listens for messages from your blocks and updates its datastore appropriately (e.g. to create a new entity)
+ * 4. It displays a debug UI allowing you to see the contents of its datastore, and messages sent back and forth
+ */
+const DevApp = () => {
+  return (
+    <MockBlockDock
+      blockDefinition={{ ReactComponent: Component }}
+      blockInfo={packageJSON.blockprotocol}
+      blockProtocolApiKey={undefined} // Set this to an API key when testing
+      blockProtocolSiteHost="https://blockprotocol-9a7200lt2.stage.hash.ai" // update this to a recent deployment when testing
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply blocks/ai-text/src/dev.tsx:26]: No – thank you

**Review by benwerner01 — APPROVED** [review]

File: blocks/ai-text/src/app/generate-text.tsx (hunk 1 above)
Comment (CiaranMn) [reply blocks/ai-text/src/app/generate-text.tsx:88]: At the moment I don't think it's worth doing this – the dependency array is almost everything that might cause a re-render in this component anyway, so I expect you'd be constructing the function on every render anyway, except with an extra step of comparing these items before you did so.

**Review by benwerner01 — APPROVED** [review]

**Pull request merged**